packages/dds/matrix/src/permutationvector.ts

```diff
@@ -199,23 +198,42 @@ export class PermutationVector extends Client {
 	}
 
 	public adjustPosition(
-		pos: number,
-		op: Pick<ISequencedDocumentMessage, "referenceSequenceNumber" | "clientId">,
-	): number | undefined {
-		const { segment, offset } = this.getContainingSegment<ISegmentInternal>(pos, {
+		posToAdjust: number,
+		op: ISequencedDocumentMessage,
+	): { pos: number | undefined; handle: Handle } | undefined {
+		const { segment, offset } = this.getContainingSegment<PermutationSegment>(posToAdjust, {
 			referenceSequenceNumber: op.referenceSequenceNumber,
 			clientId: op.clientId,
 		});
 
-		// Note that until the MergeTree GCs, the segment is still reachable via `getContainingSegment()` with
-		// a `refSeq` in the past.  Prevent remote ops from accidentally allocating or using recycled handles
-		// by checking for the presence of 'removedSeq'.
-		if (segment === undefined || segmentIsRemoved(segment)) {
-			return undefined;
-		}
+		assert(
+			segment !== undefined && offset !== undefined,
+			"segment must be available for operations in the collab window",
+		);
 
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		return this.getPosition(segment) + offset!;
+		if (segmentIsRemoved(segment)) {
+			if (!isHandleValid(segment.start)) {
+				this.applyMsg({
```

Comment:
this is a hack to split the segment. i'm looking at adding an api for this even though we shouldn't do it, ss it fragments the tree, but matrix is pretty locked into this design